examples/lorawan/Makefile

```diff
@@ -26,7 +55,40 @@ USEMODULE += $(DRIVER)
 USEMODULE += fmt
 FEATURES_REQUIRED += periph_rtc
 
-CFLAGS += -DDEVEUI=\"$(DEVEUI)\" -DAPPEUI=\"$(APPEUI)\" -DAPPKEY=\"$(APPKEY)\"
+CFLAGS += -DREGION_$(REGION)
+CFLAGS += -DSEND_PERIOD=$(SEND_PERIOD)
+ifeq (otaa,$(ACTIVATION_MODE))
+  CFLAGS += -DDEVEUI=\"$(DEVEUI)\" -DAPPEUI=\"$(APPEUI)\" -DAPPKEY=\"$(APPKEY)\"
+  CFLAGS += -DUSE_OTAA
+else ifeq (abp,$(ACTIVATION_MODE))
+  CFLAGS += -DDEVADDR=\"$(DEVADDR)\" -DNWKSKEY=\"$(NWKSKEY)\" -DAPPSKEY=\"$(APPSKEY)\"
+  CFLAGS += -DRX2_FREQ=$(RX2_FREQ) -DRX2_DR=$(RX2_DR)
+  CFLAGS += -DUSE_ABP
+endif
+
+# Enable persistence on EEPROM and lowest possible power mode only if the board
+# provides the EEPROM feature
+ifneq (,$(filter $(BOARD), $(BOARD_PROVIDING_EEPROM)))
+  FEATURES_REQUIRED += periph_eeprom
+  # Enable lowest possible power mode (e.g. STANDBY mode on STM32) which
+  # in general doesn't provide RAM retention after wake-up.
+  # This low-power mode can only be used with boards providing an internal
+  # EEPROM to be able to restore the state of MAC from a non-volatile memory.
+  CFLAGS += '-DPM_BLOCKER_INITIAL={ .val_u32 = 0x01010100 }'
+else
+  # WARNING: the next line disables the dutycycle restriction from the MAC. This
```

Comment:
Can we make this a compilation warning:

`$(warning You are disabling dutycycle restriction, make sure your sending period is than the allowed dutycycle )`

Reply:
Done